modules/library/main/src/main/java/org/geotools/data/ows/URLCheckerFactory.java

```diff
@@ -0,0 +1,119 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2020, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.ows;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+/**
+ * @author ImranR
+ *     <p>URL Checker Factory
+ *     <p>Scans for implementations of
+ *     <p>org.geotools.data.ows.URLChecker
+ */
+public class URLCheckerFactory {
+
+    // list contains all known org.geotools.data.ows.URLChecker implementations
+
+    private static List<URLChecker> urlCheckerList;
+
+    static {
+        urlCheckerList = URLCheckerFactory.initEvaluators(URLChecker.class);
+    }
+
+    private static <T> List<T> initEvaluators(Class<T> type) {
+        // allowing scanning through SDI
+        ServiceLoader<T> loader = ServiceLoader.load(type);
+        loader.reload();
+        List<T> urlCheckerList = new ArrayList<>();
+        for (T aLoader : loader) {
+            urlCheckerList.add(aLoader);
+        }
+        return urlCheckerList;
+    }
+
+    /** @return unmodifiable Collection of URLChecker implementations */
+    public static List<URLChecker> getUrlCheckerList() {

```

Comment:
Does this method need to be public?

Reply:
I use to verify the count of known implementations and also use it in unit tests